src/lib.rs

```diff
@@ -428,27 +428,35 @@ pub fn format(file: &Path, config: &Config) -> FileMap {
     file_map
 }
 
-pub fn run(file: &Path, config: &Config) {
-    let mut result = format(file, config);
+fn format_input(input: Input, config: &Config) -> (FileMap, FormatReport) {
+    let mut file_map = match input {
+        Input::File(ref file) => format(file, config),
+        Input::Text(text) => format_string(text, config),
+    };
 
-    print!("{}", fmt_lines(&mut result, config));
-    let out = stdout();
-    let write_result = filemap::write_all_files(&result, out, config);
+    let report = fmt_lines(&mut file_map, config);
+    (file_map, report)
+}
 
-    if let Err(msg) = write_result {
-        println!("Error writing files: {}", msg);
-    }
+pub enum Input {
+    File(PathBuf),
+    Text(String),
 }
 
-// Similar to run, but takes an input String instead of a file to format
-pub fn run_from_stdin(input: String, config: &Config) {
-    let mut result = format_string(input, config);
-    fmt_lines(&mut result, config);
+pub fn run(input: Input, config: &Config) {
+    let (file_map, report) = format_input(input, config);
+
+    let ignore_errors = config.write_mode == WriteMode::Plain;
```

Comment:
This roughly matches the previous behavior, but it feels ugly.

The behavior was

* swallow errors when reading from stdin (presumably because the errors and the output are written to stdout)
* print errors to stdout even with `--write-mode plain` which kinda contradicts the reasoning behind the previous bullet

Now we swallow errors every time we print to stdout. 

Note that both versions have the following bug: with `--write-mode checkstyle` errors are printed to stdout alongside the `checkstyle` xml. 


Maybe we should simply print errors to stderr? By the way, parsing errors from `libsyntax` go to the stderr anyway.